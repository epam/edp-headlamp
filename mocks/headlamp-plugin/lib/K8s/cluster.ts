export const makeKubeObject = () => {
  return class {
    static useList() {
      return [undefined, undefined];
    }

    static useGet() {
      return [undefined, undefined];
    }

    static apiEndpoint = {
      post: () => {},
    };
  };
};

export class SecretKubeObject extends makeKubeObject() {}
export class ServiceAccountKubeObject extends makeKubeObject() {}
export class CodemieKubeObject extends makeKubeObject() {}
export class CodemieProjectKubeObject extends makeKubeObject() {}
