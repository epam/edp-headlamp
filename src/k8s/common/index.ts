import { z } from 'zod';

const StringDictSchema = z.record(z.string());

const KubeOwnerReferenceSchema = z.object({
  apiVersion: z.string(),
  blockOwnerDeletion: z.boolean(),
  controller: z.boolean(),
  kind: z.string(),
  name: z.string(),
  uid: z.string(),
});

const KubeManagedFieldsEntrySchema = z.object({
  apiVersion: z.string(),
  fieldsType: z.string(),
  fieldsV1: z.object({}),
  manager: z.string(),
  operation: z.string(),
  subresource: z.string(),
  timestamp: z.string(),
});

const KubeMetadataSchema = z.object({
  annotations: StringDictSchema.optional(),
  creationTimestamp: z.string(),
  deletionGracePeriodSeconds: z.number().optional(),
  deletionTimestamp: z.string().optional(),
  finalizers: z.string().array().optional(),
  generateName: z.string().optional(),
  generation: z.number().optional(),
  labels: StringDictSchema.optional(),
  managedFields: z.array(KubeManagedFieldsEntrySchema).optional(),
  name: z.string(),
  namespace: z.string().optional(),
  ownerReferences: z.array(KubeOwnerReferenceSchema).optional(),
  resourceVersion: z.string().optional(),
  selfLink: z.string().optional(),
  uid: z.string(),
});

const KubeObjectBaseSchema = z.object({
  apiVersion: z.string(),
  kind: z.string(),
  metadata: KubeMetadataSchema,
});

const KubeObjectListBaseSchema = z.object({
  apiVersion: z.string(),
  items: z.array(KubeObjectBaseSchema),
  kind: z.string(),
  metadata: KubeMetadataSchema,
});

type KubeObjectBase = z.infer<typeof KubeObjectBaseSchema>;
type KubeObjectListBase<T extends KubeObjectBase> = z.infer<typeof KubeObjectListBaseSchema> & {
  items: T[];
};

export {
  KubeObjectBaseSchema,
  KubeMetadataSchema,
  KubeManagedFieldsEntrySchema,
  KubeOwnerReferenceSchema,
  StringDictSchema,
};

export type { KubeObjectBase, KubeObjectListBase };
