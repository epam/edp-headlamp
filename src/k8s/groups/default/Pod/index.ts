import { K8s } from '@kinvolk/headlamp-plugin/lib';

export class PodKubeObject extends K8s.pod.default {}
