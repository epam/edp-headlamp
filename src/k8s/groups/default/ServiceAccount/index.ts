import { K8s } from '@kinvolk/headlamp-plugin/lib';

export class ServiceAccountKubeObject extends K8s.serviceAccount.default {}
