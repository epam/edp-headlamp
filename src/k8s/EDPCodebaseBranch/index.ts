import { ApiProxy, K8s } from '@kinvolk/headlamp-plugin/lib';
import { STATUS_COLOR } from '../../constants/colors';
import { ICONS } from '../../icons/iconify-icons-mapping';
import { KubeObjectListInterface } from '../../types/k8s';
import { streamResults } from '../common/streamResults';
import { EDPCodebaseBranchKubeObjectConfig } from './config';
import { EDP_CODEBASE_BRANCH_STATUS } from './constants';
import { CODEBASE_BRANCH_LABEL_SELECTOR_CODEBASE_NAME } from './labels';
import {
  EDPCodebaseBranchKubeObjectInterface,
  EDPCodebaseBranchSpecInterface,
  EDPCodebaseBranchStatusInterface,
} from './types';

const {
  name: { singularForm, pluralForm },
  group,
  version,
} = EDPCodebaseBranchKubeObjectConfig;

export class EDPCodebaseBranchKubeObject extends K8s.cluster.makeKubeObject<EDPCodebaseBranchKubeObjectInterface>(
  singularForm
) {
  static apiEndpoint = ApiProxy.apiFactoryWithNamespace(group, version, pluralForm);

  static get className(): string {
    return singularForm;
  }

  static get pluralName(): string {
    return pluralForm;
  }

  get spec(): EDPCodebaseBranchSpecInterface {
    return this.jsonData!.spec;
  }

  get status(): EDPCodebaseBranchStatusInterface {
    return this.jsonData!.status;
  }

  static getStatusIcon(status: string): [string, string, boolean?] {
    if (status === undefined) {
      return [ICONS.UNKNOWN, STATUS_COLOR.UNKNOWN];
    }

    const _status = status.toLowerCase();

    switch (_status) {
      case EDP_CODEBASE_BRANCH_STATUS.CREATED:
        return [ICONS.CHECK_CIRCLE, STATUS_COLOR.SUCCESS];

      case EDP_CODEBASE_BRANCH_STATUS.FAILED:
        return [ICONS.CROSS_CIRCLE, STATUS_COLOR.ERROR];

      case EDP_CODEBASE_BRANCH_STATUS.INITIALIZED:
        return [ICONS.LOADER_CIRCLE, STATUS_COLOR.IN_PROGRESS, true];

      case EDP_CODEBASE_BRANCH_STATUS.IN_PROGRESS:
        return [ICONS.LOADER_CIRCLE, STATUS_COLOR.IN_PROGRESS, true];

      default:
        return [ICONS.UNKNOWN, STATUS_COLOR.UNKNOWN];
    }
  }

  static getListByCodebaseName(
    namespace: string,
    codebaseName: string
  ): Promise<KubeObjectListInterface<EDPCodebaseBranchKubeObjectInterface>> {
    const url = `/apis/${group}/${version}/namespaces/${namespace}/${pluralForm}?labelSelector=${CODEBASE_BRANCH_LABEL_SELECTOR_CODEBASE_NAME}=${codebaseName}`;
    return ApiProxy.request(url);
  }

  static streamListByCodebaseLabel = (
    codebaseName: string,
    cb: (data: EDPCodebaseBranchKubeObjectInterface[]) => void,
    errCb: (err: Error) => void,
    namespace: string
  ): (() => any) => {
    const url = `/apis/${group}/${version}/namespaces/${namespace}/${pluralForm}`;
    return streamResults(url, cb, errCb, {
      labelSelector: `${CODEBASE_BRANCH_LABEL_SELECTOR_CODEBASE_NAME}=${codebaseName}`,
    });
  };
}
