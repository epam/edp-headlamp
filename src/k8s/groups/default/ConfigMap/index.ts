import { K8s } from '@kinvolk/headlamp-plugin/lib';
import { streamResults } from '../../../common/streamResults';
import { ConfigMapKubeObjectConfig } from './config';
import { StreamListProps } from './types';

const {
  name: { pluralForm },
  version,
} = ConfigMapKubeObjectConfig;

export class ConfigMapKubeObject extends K8s.configMap.default {
  static streamList({ namespace, dataHandler, errorHandler }: StreamListProps): () => void {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}`;
    return streamResults(url, dataHandler, errorHandler);
  }
}
