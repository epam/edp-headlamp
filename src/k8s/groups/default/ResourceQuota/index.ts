import { K8s } from '@kinvolk/headlamp-plugin/lib';
import { streamResults } from '../../../common/streamResults';
import { ResourceQuotaKubeObjectConfig } from './config';
import { RESOURCE_QUOTA_LABEL_TENANT } from './labels';
import { StreamListProps } from './types';

const {
  name: { pluralForm },
  version,
} = ResourceQuotaKubeObjectConfig;

export class ResourceQuotaKubeObject extends K8s.ResourceClasses.ResourceQuota {
  static streamList({
    namespace,
    tenantNamespace,
    dataHandler,
    errorHandler,
  }: StreamListProps): () => void {

    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}`;
    return streamResults(url, dataHandler, errorHandler, {
      labelSelector: `${RESOURCE_QUOTA_LABEL_TENANT}=edp-workload-${tenantNamespace}`,
    });
  }
}
