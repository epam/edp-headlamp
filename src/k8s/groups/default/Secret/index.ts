import { ApiProxy, K8s } from '@kinvolk/headlamp-plugin/lib';
import { STATUS_COLOR } from '../../../../constants/colors';
import { ICONS } from '../../../../icons/iconify-icons-mapping';
import { KubeObjectListInterface } from '../../../../types/k8s';
import { SecretKubeObjectConfig } from './config';
import { SECRET_LABEL_SECRET_TYPE } from './labels';
import { SecretKubeObjectInterface } from './types';

const {
  name: { pluralForm },
  version,
} = SecretKubeObjectConfig;

export class SecretKubeObject extends K8s.secret.default {
  static getClusterSecretList(
    namespace: string
  ): Promise<KubeObjectListInterface<SecretKubeObjectInterface>> {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}?labelSelector=${SECRET_LABEL_SECRET_TYPE}=cluster`;

    return ApiProxy.request(url);
  }

  static getSecretByName(namespace: string, name: string): Promise<SecretKubeObjectInterface> {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}/${name}`;

    return ApiProxy.request(url);
  }

  static getStatusIcon(connected: string): [string, string, boolean?] {
    if (connected === undefined) {
      return [ICONS.UNKNOWN, STATUS_COLOR.UNKNOWN];
    }

    if (connected === 'true') {
      return [ICONS.CHECK_CIRCLE, STATUS_COLOR.SUCCESS];
    }
    return [ICONS.CROSS_CIRCLE, STATUS_COLOR.ERROR];
  }
}
