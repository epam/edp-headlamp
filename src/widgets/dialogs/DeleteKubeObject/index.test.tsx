/**
 * @jest-environment jsdom
 */

import { KubeObjectInterface } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../../mocks/wrappers/default';
import { DeleteKubeObjectDialog } from './index';

test('renders DeleteKubeObjectDialog component', () => {
  render(
    <TestWrapper>
      <DeleteKubeObjectDialog
        props={{
          onSuccess: jest.fn(),
          objectName: 'MockObjectName',
          kubeObjectData: {
            kind: 'MockKind',
          } as KubeObjectInterface,
          kubeObject: {},
          onBeforeSubmit: jest.fn(),
          description: 'Mock Description',
        }}
        state={{
          open: true,
          openDialog: jest.fn(),
          closeDialog: jest.fn(),
        }}
      />
    </TestWrapper>
  );

  // Check that the component is rendered
  const dialog = screen.getByTestId('dialog');
  expect(dialog).toMatchSnapshot();
});
