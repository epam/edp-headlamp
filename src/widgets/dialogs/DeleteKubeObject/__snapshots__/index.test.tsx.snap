// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DeleteKubeObjectDialog component 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="mui-1"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-15hyav-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-8yphvn-MuiTypography-root-MuiDialogTitle-root"
        id="mui-1"
      >
        Confirm deletion of "MockObjectName"
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1ezega9-MuiTypography-root"
            >
              Mock Description
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <form>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-8dtgre-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="mui-2"
                      id="mui-2-label"
                    >
                      Enter "MockObjectName" to delete
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-71czni-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-2"
                        name="name"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Enter "MockObjectName" to delete
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                >
                  <div
                    class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1wnvitu-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Cancel
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1wnvitu-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="submit"
                    >
                      Confirm
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
